private-templates-service/client/src/components/Dashboard/PreviewModal/TemplateInfo/TemplateInfo.tsx

```diff
@@ -98,10 +98,23 @@ class TemplateInfo extends React.Component<Props, State> {
     this.state = { isPublishOpen: false, version: vers }
   }
 
+  redirectToDesigner = () => {
+    const { history } = this.props;
+    if (history) history.push('/designer');
+  }
+
   toggleModal = () => {
     this.setState({ isPublishOpen: !this.state.isPublishOpen });
   }
 
+  handleClick = (val: string) => {
+    if (val == 'Publish') {
+      this.toggleModal();
+    } else if (val == 'Edit in designer') {
+      this.redirectToDesigner();
```

Comment:
I did something very similar in my PR, it's okay to be merged into dev right now but I think this will change with my PR, just letting you know